tools/html_transformer/html_transformer.js

```diff
@@ -0,0 +1,95 @@
+var Promise = require('es6-promise').Promise;
+var jsdom = require('jsdom');
+
+exports.window = null;
+
+exports.load = function(filename) {
+  return new Promise(function(resolve, reject) {
+    jsdom.env(filename, function(errors, window) {
+      if (errors && errors.length) {
+        return reject(errors[0]);
+      }
+
+      exports.window = window;
+      return resolve();
```

Comment:
I don't know if it's just me, but this feels a little weird. What if instead of resolving with `window` on `exports` asynchronously, you return an instance of an object that refers to the window, etc. Example:

```js
var Transformer = function(window) {
  this.window = window;
};

Transformer.prototype.getSharedDependencies = function() {};

exports.load = function(filename) {
  // blah, blah, blah
  return resolve(new Transformer(window));
};
```